deploy/aws.rst

```diff
@@ -0,0 +1,107 @@
+      }
+    }
+  }
+
+  resource "aws_iam_role" "saml_role" {
+    name               = "IdP-<some name>"
+    assume_role_policy = data.aws_iam_policy_document.saml_policy.json
+  }
+
+The name of the role needs to be the name of the moira list prefixed by ``IdP-``. For example, if the moira list you created was called ``aws-672626379771-foobar``, then the AWS role name should be ``IdP-foobar``.
+
+This new role can now be used as you would any other IAM role. Users on the moira list who need to access AWS should log in the same way everyone does, through https://aws.mit.edu. If they are on more than one of these special moria lists, they will be presented with the option to choose which role they want to log in as. You can only be logged in under one of these roles at a time.
+
+Weird Things
+------------
+
+Sometimes AWS is weird and frustrating. These things go here.
+
+RSA 4096 Certs
+~~~~~~~~~~~~~~
+
+Amazon's Certificate Manager does not handle RSA 4096 certificates yet. If you have one of these it will need to be added to IAM. There is no way that I can find to do this through either Terraform or the web console. You'll have to use the AWS CLI::
+
+  $ aws iam upload-server-certificate \
+    --server-certificate-name <hostname> \
+    --certificate-body file://<your cert>.crt \
+    --certificate-chain file://<your cert>.chain.crt \
+    --private-key file://<your cert>.key
+
+EFS Mount Race Conditions
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+I have observed on multiple occasions problems with mounting newly created EFS mounts in EC2 instances. This seems more likely to happen with Terraform due to resources being provisioned all at once. My guess is the DNS for the new EFS mount has not propagated by the time the cloud-init script is run, where you would usually do the NFS mount.
+
+There's no obvious (easy) mitigation for this. The good news is since it appears to simply be a DNS propagation issue, this should only be a problem for a short period after the initial provisioning of the EFS mount. My suggestion is to check the logs for your cloud-init script if you are spinning up an EC2 instance at the same time you are creating the EFS mount. There should be an error in there if it can't resolve the hostname for the mount.
+
+Fargate Log Flushing
+~~~~~~~~~~~~~~~~~~~~
+
+Sometimes logs in Fargate don't get written, or are only partially written, to Cloudwatch. My own experience suggests the problem is that the logs are being discarded before being fully flushed to Cloudwatch. The fix, which I have found to be reliable, is to add a few seconds of sleep to your container after you have stopped the main process. You can see an example here: https://github.com/MITLibraries/workflow/blob/master/entrypoint.sh.
+
+S3 Bucket Limit
+~~~~~~~~~~~~~~~
+
+There's a limit to the number of buckets an account can have in S3. Rather than creating a bunch of buckets, partition a few buckets with predictable prefixes.
```

Comment:
I'm not sure if it's worth noting that while this is a good idea, we have not done this to date and thus this is a "do as I say, don't look at what we have done so far for examples" kinda thing.